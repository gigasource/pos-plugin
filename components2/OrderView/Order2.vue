<template>
  <div class="order">
    <div class="order-main">
      <pos-order-layout2 style="flex: 1"
                         v-model:selectedCategoryLayout="selectedCategoryLayout"
                         v-model:selectedProductLayout="selectedProductLayout"
                         v-model:productDblClicked="productDblClicked"
                         v-model:orderLayout="orderLayout"
                         v-model:keyboardConfig="keyboardConfig"
                         v-model:fontSize="fontSize"
                         v-model:category="category"
                         v-model:minimumTextRow="minimumTextRow"
                         v-model:collapseBlankColumn="collapseBlankColumn"
                         v-model:hideTextRow="hideTextRow"
                         v-model:hideBlankColumn="hideBlankColumn"
                         v-model:collapseText="collapseText"
                         v-model:actionMode="actionMode"
                         v-model:showOverlay="showOverlay"
                         v-model:scrollableLayout="scrollableLayout"
                         :is-mobile="isMobile"
                         :store-locale="storeLocale"
                         @addProductToOrder="addProductToOrder"
                         @addModifierToProduct="addModifierToProduct"
                         @getKeyboardSetting="getKeyboardSetting"
      />
      <pos-order :style="!isMobile && {flex: '0 0 25%'}"
                 v-model:fontSize="fontSize"
                 v-model:category="category"
                 v-model:minimumTextRow="minimumTextRow"
                 v-model:collapseBlankColumn="collapseBlankColumn"
                 v-model:hideTextRow="hideTextRow"
                 v-model:hideBlankColumn="hideBlankColumn"
                 v-model:collapseText="collapseText"
                 v-model:actionMode="actionMode"
                 v-model:showOverlay="showOverlay"
                 v-model:scrollableLayout="scrollableLayout"
                 :is-mobile="isMobile"
                 :store-locale="storeLocale"
                 :user="user"
                 :items="currentOrder.items"
                 :total="paymentTotal"
                 :printedOrder="printedOrder"
                 :actionList="actionList"
                 :isTakeAwayOrder="isTakeAwayOrder"
                 @addItemQuantity="addItemQuantity"
                 @removeItemQuantity="removeItemQuantity"
                 @addModifierToProduct="addModifierToProduct"
                 @changePrice="changePrice"
                 @updateOrderTable="updateOrderTable"
                 @updateOrderItems="updateOrderItems"
                 @removeProductModifier="removeProductModifier"
                 @updateOrderItem="updateOrderItem"
                 @saveTableOrder="saveTableOrder"
                 @updateCurrentOrder="updateCurrentOrder"
                 @resetOrderData="resetOrderData"
                 @quickCash="quickCashRestaurant"
                 @openDialog="openDialog"
      />
    </div>
    <pos-quick-order-toolbar
        v-if="!isMobile"
        :currentOrder="currentOrder"
        :actionList="actionList"
        @updateCurrentOrder="updateCurrentOrder"
        @resetOrderData="resetOrderData"
        @quickCash="quickCashRestaurant"
        @saveTableOrder="saveTableOrder"
        @openDialog="openDialog"
    />
  </div>
  <pos-order-split-order
      v-model="dialog.split"
      :is-mobile="isMobile"
      :user="user"
      :store-locale="storeLocale"
      :current-order="currentOrder"
      @updateCurrentOrder="updateCurrentOrder"
      @createOrderCommit="createOrderCommit"
      @saveSplitOrder="saveSplitOrder"
      @resetOrderData="resetOrderData"
      @printOrderReport="printOrderReport"
      @updatePrintedOrder="updatePrintedOrder"
  />
  <pos-order-receipt
      v-model="dialog.receipt"
      :order="currentOrder"
      :store-locale="storeLocale"
  />
  <pos-order-move-items
      v-model="dialog.move"
      :is-mobile="isMobile"
      :user="user"
      :store-locale="storeLocale"
      :current-order="currentOrder"
      :active-orders="activeOrders"
      @moveItems="moveItems"
  />
  <pos-order-voucher-dialog
      v-model="dialog.voucher"
      @addVoucher="addVoucher"
      @redeemVoucher="redeemVoucher"
  />
</template>

<script>

import PosOrder2 from "./PosOrder2";
import PosOrderLayout2 from "./PosOrderLayout2";
import PosQuickOrderToolbar2 from "./Helper/PosQuickOrderToolbar2";
import {isMobile} from "../AppSharedStates";

export default {
  name: 'Order2',
  components: {
    PosOrder2,
    PosOrderLayout2,
    PosQuickOrderToolbar2,
  },
  setup() {
    //todo: put style in components

    const render = () => (<>
      <div class="order">
        <div class="order-main">
          <pos-order-layout2 style="flex: 1"/>
          <pos-order style={!isMobile.value && {flex: '0 0 25%'}}/>
        </div>
        <pos-quick-order-toolbar
            v-if="!isMobile"
            currentOrder="currentOrder"
            actionList={actionList}
            updateCurrentOrder={updateCurrentOrder}
            resetOrderData={resetOrderData}
            quickCash={quickCashRestaurant}
            saveTableOrder={saveTableOrder}
            openDialog={openDialog}
        />
      </div>
      <pos-order-split-order
          v-model={dialog.split}
          is-mobile={isMobile}
          user={user}
          store-locale={storeLocale}
          current-order={currentOrder}
          updateCurrentOrder={updateCurrentOrder}
          createOrderCommit={createOrderCommit}
          saveSplitOrder={saveSplitOrder}
          resetOrderData={resetOrderData}
          printOrderReport={printOrderReport}
          updatePrintedOrder={updatePrintedOrder}
      />
      <pos-order-receipt
          v-model={dialog.receipt}
      />
      <pos-order-move-items
          v-model={dialog.move}
          user={user}
          current-order={currentOrder}
          active-orders={activeOrders}
          moveItems={moveItems}
      />
      <pos-order-voucher-dialog
          v-model={dialog.voucher}
          addVoucher={addVoucher}
          redeemVoucher={redeemVoucher}
      />
    </>)
  },
  data() {
    return {
      orderLayout: null,
      selectedCategoryLayout: null,
      selectedProductLayout: null,
      productDblClicked: null,
      keyboardConfig: null,
      fontSize: '14px',
      category: {
        type: 'horizontal',
        size: '64px',
        singleRow: false,
        differentSize: false,
        fontSize: '13px'
      },
      minimumTextRow: true,
      collapseBlankColumn: true,
      hideTextRow: false,
      hideBlankColumn: false,
      collapseText: true,
      actionMode: 'none',
      showOverlay: true,
      scrollableLayout: true,
      //inject
      user: {},
      isMobile: null,
      storeLocale: null,
      currentOrder: {},
      paymentTotal: null,
      printedOrder: null,
      actionList: [],
      isTakeAwayOrder: null,
      activeOrders: null,
      //dialog
      dialog: {
        search: false,
        split: false,
        move: false,
        voucher: false,
        receipt: false
      }
    }
  },
  methods: {
    addProductToOrder() {
    },
    addModifierToProduct() {
    },
    getKeyboardSetting() {
    },
    addItemQuantity() {
    },
    removeItemQuantity() {
    },
    changePrice() {
    },
    updateOrderTable() {
    },
    updateOrderItems() {
    },
    removeProductModifier() {
    },
    updateOrderItem() {
    },
    saveTableOrder() {
    },
    resetOrderData() {
    },
    updateCurrentOrder() {
    },
    quickCashRestaurant() {
    },
    openDialog(name) {
      this.dialog[name] = true
    },
    createOrderCommit() {
    },
    saveSplitOrder() {
    },
    printOrderReport() {
    },
    updatePrintedOrder() {
    },
    moveItems() {
    },
    addVoucher() {
    },
    redeemVoucher() {
    }
  }
}
</script>

<style scoped lang="scss">
.order {
  height: 100%;
  display: flex;
  flex-direction: column;

  .order-main {
    flex: 1;
    display: flex;
  }

  .g-toolbar {
    flex: 0 1 auto;
  }
}
</style>
