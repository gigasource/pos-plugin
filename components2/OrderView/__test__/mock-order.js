const mockOrder = {
  table: "10",
  _id: "5fee6600612d45b5ffdcabb7",
  items:
    [{
      groupPrinter: "Bar",
      id: "48",
      name: "Whiskey",
      price: 40,
      product: "5e784d7786e3ec0997b5e39b",
      taxes: [19, 7],
      course: 1,
      quantity: 1,
      _id: "5fee6600612d45b5ffdcabb8",
      vSum: 40,
      vTakeAway: false,
      tax: 19,
      sent: true,
      printed: true,
      vTaxSum: {
        "19": {
          tax: 6.39,
          net: 33.61,
          gross: 40
        }
      }
    }],
  cancellationItems: [],
  takeAway: false,
  status: "inProgress",
  payment: [],
  user: [],
  vSum: 40,
  vDate: "2020-12-31T17:00:00.000Z",
  vTaxSum: {
    '19': {
      tax: 6.39,
      net: 33.61,
      gross: 40
    }
  }
}

module.exports = {
  mockOrder
}
