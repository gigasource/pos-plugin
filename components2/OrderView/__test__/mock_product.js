export const mockProduct = {
  "_id": "5e784d7786e3ec0997b5e39b",
  "category": {
    "_id": "5df6f5dd7854273fa4fd989e",
    "name": "Wine",
    "position": 5
  },
  "groupPrinter": {
    "_id": "5e7864b186e3ec0997b5e48e",
    "name": "Bar",
    "printer": "HP M176n",
    "printers": [
      {
        "_id": "5e81774685fb2d03dda95cd0",
        "printerType": "ip",
        "ip": "192.168.10.6",
        "escPOS": false,
        "merge": "true"
      }
    ],
    "showDineInTax": false,
    "type": "kitchen"
  },
  "id": "48",
  "layouts": [
    {
      "_id": "5e784d7786e3ec0997b5e39c",
      "color": "#DCE775",
      "order": 2,
      "device": "pos"
    }
  ],
  "name": "Whiskey",
  "option": {},
  "price": 40,
  "tax": 19,
  "tax2": 7
}
