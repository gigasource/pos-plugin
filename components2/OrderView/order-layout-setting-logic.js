import {onActivated, reactive, ref} from "vue";

export const fontSize = ref('14px');
export const category = reactive({
  type: 'horizontal', size: '64px', singleRow: false,
  differentSize: false, fontSize: '13px'
});
export const minimumTextRow = ref(true);
export const collapseBlankColumn = ref(true);
export const collapseText = ref(true);
export const showOverlay = ref(true);
export const showSplitBtn = ref(true);
export const smallSidebar = ref(true);
export const scrollableLayout = ref(true);
export const editModeOL = ref(false);
export const hideTextRow = ref(false);
export const hideBlankColumn = ref(false);

const olSetting = reactive({
  fontSize, category, minimumTextRow, collapseBlankColumn,
  collapseText, showOverlay, showSplitBtn, smallSidebar, scrollableLayout,
  hideTextRow, hideBlankColumn
})

export function saveOrderLayoutSetting() {
  const setting = {
    fontSize, category, minimumTextRow, collapseBlankColumn,
    collapseText, showOverlay, showSplitBtn, smallSidebar, scrollableLayout,
    hideTextRow, hideBlankColumn
  }
  localStorage.setItem('OrderScreenSetting', JSON.stringify(setting))
}

export function loadOrderLayoutSetting() {
  let setting = localStorage.getItem('OrderScreenSetting')
  if (setting) {
    setting = JSON.parse(setting);
    Object.keys(olSetting)
    for (const key of Object.keys(olSetting)) {
      if (key === 'category') {
        Object.assign(category, setting[key]);
        continue
      }
      olSetting[key] = setting[key];
    }
  }
}

loadOrderLayoutSetting();

export function autoLoadOrderLayoutSetting() {
  onActivated(() => {
    loadOrderLayoutSetting();
  })
}

export function changeSize(num) {
  const size = +fontSize.value.slice(0, fontSize.value.length - 2)
  if (size + num > 0) {
    fontSize.value = `${size + num}px`
  }
}

export function changeCategory() {
  category.type = category.type === 'horizontal' ? 'vertical' : 'horizontal';
}

export function changeCategorySize(change) {
  const size = +category.size.slice(0, category.size.length - 2)
  if (size + change > 0) {
    Object.assign(category, {size: `${size + change}px`})
  }
}

export function changeCategoryFontSize(change) {
  const size = +category.fontSize.slice(0, category.fontSize.length - 2)
  if (size + change > 0) {
    category.fontSize = `${size + change}px`;
  }
}

export function changeCategoryStyle(key, value) {
  let change = {}
  change[key] = value
  Object.assign(category, change);
}
