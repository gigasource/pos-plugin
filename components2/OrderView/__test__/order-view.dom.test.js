//<editor-fold desc="declare">
import {
  actions,
  cola,
  drinkTax,
  expectArray,
  fanta,
  foodTax,
  ketchup,
  rice,
  component,
  wrapper,
  makeWrapper,
  orm,
  setComponent
} from "../../test-utils";
import { config, mount } from "@vue/test-utils";

import { nextTick } from "vue";
import { payPrintBtnFactory } from "../payPrintBtnFactory";

import {
  addProduct,
  disablePay,
  getCurrentOrder,
  payBtnClickable,
  payPrintMode,
  prepareOrder,
  showIcon,
  togglePayPrintBtn
} from "../pos-logic-be";

import { mockProduct } from "./mock_product";
import { mockOrder } from "./mock-order";
import { smallSidebar } from "../order-layout-setting-logic";
import { orderRightSideHeader } from "../orderRightSideHeaderFactory";
import { orderRightSideItemsTable } from "../orderRightSideItemsTable";
import PosOrder2 from "../PosOrder2";
import Order2 from "../Order2";
import { orderLayout } from "../pos-ui-shared";
import { demoData } from "./demoData";
import PosOrderLayout2 from "../PosOrderLayout2";

const { stringify } = require("schemahandler/utils");

const {
  makeDiscount,
  makeTakeaway,
  addModifier,
  addItem,
  createOrder,
  makeLastItemDiscount
} = require("../pos-logic");

const delay = require("delay");

//</editor-fold>

describe("order-view test", function() {
  beforeAll(async () => {
    await orm("PosSetting").remove({});
    await orm("PosSetting").create(demoData.PosSetting[0]);
    await orm("OrderLayout").remove({});
    await orm("OrderLayout").create(demoData.OrderLayout[0]);
  });
  it("case 1 integration", async function() {
    //order have 1 sent item, add one item -> should display print,
    prepareOrder(mockOrder);
    let order = getCurrentOrder();
    addProduct(order, mockProduct);
    orderLayout.value = demoData.OrderLayout[0];
    makeWrapper(Order2, { shallow: false });
    await nextTick();
    await delay(50);
    expect(wrapper.html()).toMatchSnapshot();
  }, 80000);

  it("case 2 orderLayout", async function() {
    //order have 1 sent item, add one item -> should display print,
    prepareOrder(mockOrder);
    let order = getCurrentOrder();
    addProduct(order, mockProduct);
    orderLayout.value = demoData.OrderLayout[0];
    makeWrapper(PosOrderLayout2, { shallow: true });
    await nextTick();
    await delay(50);
    expect(wrapper.html()).toMatchSnapshot()
  }, 80000);


});
