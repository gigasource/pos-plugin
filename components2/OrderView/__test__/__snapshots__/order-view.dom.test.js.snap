// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`order-view test case 1 integration 1`] = `
<div class="order">
  <div class="order-main">
    <div class="pol"
         style="flex-direction: column; background-size: contain; overflow: hidden; background: url(/plugins/pos-plugin/assets/out.png);"
    >
      <div style="padding: 4px; background-color: rgb(224, 224, 224); position: sticky; top: 0px; z-index: 1;">
        <div>
          <div class="pol__cate"
               style="background-color: rgb(178, 235, 242); color: rgb(0, 0, 0); box-shadow: none; grid-area: 1 / 1 / 2 / 2; border: 1px solid transparent;"
          >
            Drink
          </div>
        </div>
      </div>
      <div style="padding: 4px; flex: 1;">
        <div>
          <div class="pol__prod darken-effect"
               style="grid-area: 1 / 1 / 2 / 2; background-color: transparent; color: rgb(33, 33, 33); border-radius: 2px; font-weight: 400; line-height: 1.2;"
          >
            <div>
              Soft Drink
            </div>
          </div>
          <div class="pol__prod darken-effect"
               style="grid-area: 2 / 1 / 3 / 2; background-color: rgb(178, 235, 242); color: rgb(0, 0, 0); border-radius: 2px;"
          >
            <div>
            </div>
          </div>
          <div class="pol__prod darken-effect"
               style="grid-area: 1 / 2 / 2 / 3; background-color: transparent; color: rgb(33, 33, 33); border-radius: 2px; font-weight: 400; line-height: 1.2;"
          >
            <div>
              Wine
            </div>
          </div>
          <div class="pol__prod darken-effect"
               style="grid-area: 2 / 2 / 3 / 3; background-color: rgb(200, 230, 201); color: rgb(0, 0, 0); border-radius: 2px;"
          >
            <div>
            </div>
          </div>
          <div class="pol__prod darken-effect"
               style="grid-area: 3 / 2 / 4 / 3; background-color: rgb(200, 230, 201); color: rgb(0, 0, 0); border-radius: 2px;"
          >
            <div>
            </div>
          </div>
          <div class="pol__prod darken-effect"
               style="grid-area: 4 / 2 / 5 / 3; background-color: rgb(200, 230, 201); color: rgb(0, 0, 0); border-radius: 2px;"
          >
            <div>
            </div>
          </div>
          <div class="pol__prod darken-effect"
               style="grid-area: 5 / 2 / 6 / 3; background-color: rgb(200, 230, 201); color: rgb(0, 0, 0); border-radius: 2px;"
          >
            <div>
            </div>
          </div>
          <div class="pol__prod darken-effect"
               style="grid-area: 2 / 3 / 3 / 4; background-color: rgb(200, 230, 201); color: rgb(0, 0, 0); border-radius: 2px;"
          >
            <div>
            </div>
          </div>
          <div class="pol__prod darken-effect"
               style="grid-area: 3 / 3 / 4 / 4; background-color: rgb(200, 230, 201); color: rgb(0, 0, 0); border-radius: 2px;"
          >
            <div>
            </div>
          </div>
          <div class="pol__prod darken-effect"
               style="grid-area: 4 / 3 / 5 / 4; background-color: rgb(200, 230, 201); color: rgb(0, 0, 0); border-radius: 2px;"
          >
            <div>
            </div>
          </div>
          <div class="pol__prod darken-effect"
               style="grid-area: 5 / 3 / 6 / 4; background-color: rgb(200, 230, 201); color: rgb(0, 0, 0); border-radius: 2px;"
          >
            <div>
            </div>
          </div>
          <div class="pol__prod darken-effect"
               style="grid-area: 2 / 4 / 3 / 5; background-color: rgb(200, 230, 201); color: rgb(0, 0, 0); border-radius: 2px;"
          >
            <div>
            </div>
          </div>
          <div class="pol__prod darken-effect"
               style="grid-area: 6 / 1 / 7 / 2; background-color: transparent; color: rgb(33, 33, 33); border-radius: 2px; font-weight: 400; line-height: 1.2;"
          >
            <div>
              Other
            </div>
          </div>
          <div class="pol__prod darken-effect"
               style="grid-area: 7 / 1 / 8 / 2; background-color: rgb(255, 245, 157); color: rgb(0, 0, 0); border-radius: 2px;"
          >
            <div>
            </div>
          </div>
          <div class="pol__prod darken-effect"
               style="grid-area: 8 / 1 / 9 / 2; background-color: rgb(255, 245, 157); color: rgb(0, 0, 0); border-radius: 2px;"
          >
            <div>
            </div>
          </div>
          <div class="pol__prod darken-effect"
               style="grid-area: 9 / 1 / 10 / 2; background-color: rgb(255, 245, 157); color: rgb(0, 0, 0); border-radius: 2px;"
          >
            <div>
            </div>
          </div>
          <div class="pol__prod darken-effect"
               style="grid-area: 10 / 1 / 11 / 2; background-color: rgb(255, 245, 157); color: rgb(0, 0, 0); border-radius: 2px;"
          >
            <div>
            </div>
          </div>
          <div class="pol__prod darken-effect"
               style="grid-area: 1 / 6 / 2 / 7; background-color: transparent; color: rgb(33, 33, 33); border-radius: 2px; font-weight: 400; line-height: 1.2;"
          >
            <div>
              Topping
            </div>
          </div>
          <div class="pol__prod darken-effect"
               style="grid-area: 2 / 6 / 3 / 7; background-color: rgb(206, 147, 216); color: rgb(0, 0, 0); border-radius: 2px;"
          >
            <div>
            </div>
          </div>
          <div class="pol__prod darken-effect"
               style="grid-area: 3 / 6 / 4 / 7; background-color: rgb(206, 147, 216); color: rgb(0, 0, 0); border-radius: 2px;"
          >
            <div>
            </div>
          </div>
          <div class="pol__prod darken-effect"
               style="grid-area: 4 / 6 / 5 / 7; background-color: rgb(206, 147, 216); color: rgb(0, 0, 0); border-radius: 2px;"
          >
            <div>
            </div>
          </div>
          <div class="pol__prod darken-effect"
               style="grid-area: 8 / 2 / 9 / 3; background-color: rgb(255, 171, 145); color: rgb(0, 0, 0); border-radius: 2px;"
          >
            <div>
            </div>
          </div>
          <div class="pol__prod darken-effect"
               style="grid-area: 9 / 2 / 10 / 3; background-color: rgb(255, 171, 145); color: rgb(0, 0, 0); border-radius: 2px;"
          >
            <div>
            </div>
          </div>
          <div class="pol__prod darken-effect"
               style="grid-area: 10 / 2 / 11 / 3; background-color: rgb(255, 171, 145); color: rgb(0, 0, 0); border-radius: 2px;"
          >
            <div>
            </div>
          </div>
          <pos-order-keyboard-stub>
          </pos-order-keyboard-stub>
          <dialog-text-filter-stub>
          </dialog-text-filter-stub>
          <dialog-product-search-result-stub>
          </dialog-product-search-result-stub>
        </div>
      </div>
      <dialog-choose-popup-modifier-stub>
      </dialog-choose-popup-modifier-stub>
      <g-overlay-stub value="false"
                      absolute="true"
                      opacity="0.25"
                      color="rgb(150, 150, 150)"
      >
        <g-icon-stub size="120">
          icon-print
        </g-icon-stub>
      </g-overlay-stub>
    </div>
    <pos-order-stub>
    </pos-order-stub>
  </div>
</div>
<div>
  <g-dialog-stub modelvalue="false"
                 transition="false"
                 content-class="move-items-dialog"
  >
    <div class="row-flex justify-end w-100">
      <div class="splitter"
           style="height: calc(100% - 84px);"
      >
        <div class="splitter__header row-flex align-items-center">
          0
          <g-spacer-stub>
          </g-spacer-stub>
          <g-btn-bs-stub uppercase="false"
                         disabled="true"
                         icon="icon-move-items"
                         background-color="#1271ff"
          >
            <span>
              Move Items
            </span>
          </g-btn-bs-stub>
        </div>
        <div class="splitter__content">
        </div>
      </div>
      <g-icon-stub class="mr-4 ml-4"
                   size="40"
                   color="#fff"
                   style="height: calc(100% - 64px);"
      >
        keyboard_backspace
      </g-icon-stub>
      <div class="order-detail"
           style="height: calc(100% - 64px);"
      >
        <div class="order-detail__header row-flex align-items-center">
          <div class="row-flex align-items-center">
            <g-avatar-stub size="36">
              <img alt="true">
            </g-avatar-stub>
            <div class="ml-1 fw-700"
                 style="font-size: 13px;"
            >
              <div>
              </div>
              <div>
                <span style="font-weight: 600; font-size: 11px; color: rgb(158, 158, 158);">
                  Table
                </span>
                <span class="ml-2"
                      style="font-weight: 600; font-size: 13px;"
                >
                  10
                </span>
              </div>
            </div>
          </div>
          <g-spacer-stub>
          </g-spacer-stub>
        </div>
        <div class="order-detail__content">
        </div>
      </div>
    </div>
    <g-toolbar-stub elevation="0"
                    color="#eee"
                    class="toolbar"
    >
      <g-btn-bs-stub icon="icon-back">
        ui.back
      </g-btn-bs-stub>
      <g-spacer-stub>
      </g-spacer-stub>
    </g-toolbar-stub>
  </g-dialog-stub>
  <choose-table-dialog-stub table="10"
                            active-orders
                            modelvalue="false"
  >
  </choose-table-dialog-stub>
</div>
`;

exports[`order-view test case 2 orderLayout 1`] = `
<div class="pol"
     style="flex-direction: column; background-size: contain; overflow: hidden; background: url(/plugins/pos-plugin/assets/out.png);"
>
  <div style="padding: 4px; background-color: rgb(224, 224, 224); position: sticky; top: 0px; z-index: 1;">
    <div>
      <div class="pol__cate"
           style="background-color: rgb(178, 235, 242); color: rgb(0, 0, 0); box-shadow: none; grid-area: 1 / 1 / 2 / 2; border: 1px solid transparent;"
      >
        Drink
      </div>
    </div>
  </div>
  <div style="padding: 4px; flex: 1;">
    <div>
      <div class="pol__prod darken-effect"
           style="grid-area: 1 / 1 / 2 / 2; background-color: transparent; color: rgb(33, 33, 33); border-radius: 2px; font-weight: 400; line-height: 1.2;"
      >
        <div>
          Soft Drink
        </div>
      </div>
      <div class="pol__prod darken-effect"
           style="grid-area: 2 / 1 / 3 / 2; background-color: rgb(178, 235, 242); color: rgb(0, 0, 0); border-radius: 2px;"
      >
        <div>
        </div>
      </div>
      <div class="pol__prod darken-effect"
           style="grid-area: 1 / 2 / 2 / 3; background-color: transparent; color: rgb(33, 33, 33); border-radius: 2px; font-weight: 400; line-height: 1.2;"
      >
        <div>
          Wine
        </div>
      </div>
      <div class="pol__prod darken-effect"
           style="grid-area: 2 / 2 / 3 / 3; background-color: rgb(200, 230, 201); color: rgb(0, 0, 0); border-radius: 2px;"
      >
        <div>
        </div>
      </div>
      <div class="pol__prod darken-effect"
           style="grid-area: 3 / 2 / 4 / 3; background-color: rgb(200, 230, 201); color: rgb(0, 0, 0); border-radius: 2px;"
      >
        <div>
        </div>
      </div>
      <div class="pol__prod darken-effect"
           style="grid-area: 4 / 2 / 5 / 3; background-color: rgb(200, 230, 201); color: rgb(0, 0, 0); border-radius: 2px;"
      >
        <div>
        </div>
      </div>
      <div class="pol__prod darken-effect"
           style="grid-area: 5 / 2 / 6 / 3; background-color: rgb(200, 230, 201); color: rgb(0, 0, 0); border-radius: 2px;"
      >
        <div>
        </div>
      </div>
      <div class="pol__prod darken-effect"
           style="grid-area: 2 / 3 / 3 / 4; background-color: rgb(200, 230, 201); color: rgb(0, 0, 0); border-radius: 2px;"
      >
        <div>
        </div>
      </div>
      <div class="pol__prod darken-effect"
           style="grid-area: 3 / 3 / 4 / 4; background-color: rgb(200, 230, 201); color: rgb(0, 0, 0); border-radius: 2px;"
      >
        <div>
        </div>
      </div>
      <div class="pol__prod darken-effect"
           style="grid-area: 4 / 3 / 5 / 4; background-color: rgb(200, 230, 201); color: rgb(0, 0, 0); border-radius: 2px;"
      >
        <div>
        </div>
      </div>
      <div class="pol__prod darken-effect"
           style="grid-area: 5 / 3 / 6 / 4; background-color: rgb(200, 230, 201); color: rgb(0, 0, 0); border-radius: 2px;"
      >
        <div>
        </div>
      </div>
      <div class="pol__prod darken-effect"
           style="grid-area: 2 / 4 / 3 / 5; background-color: rgb(200, 230, 201); color: rgb(0, 0, 0); border-radius: 2px;"
      >
        <div>
        </div>
      </div>
      <div class="pol__prod darken-effect"
           style="grid-area: 6 / 1 / 7 / 2; background-color: transparent; color: rgb(33, 33, 33); border-radius: 2px; font-weight: 400; line-height: 1.2;"
      >
        <div>
          Other
        </div>
      </div>
      <div class="pol__prod darken-effect"
           style="grid-area: 7 / 1 / 8 / 2; background-color: rgb(255, 245, 157); color: rgb(0, 0, 0); border-radius: 2px;"
      >
        <div>
        </div>
      </div>
      <div class="pol__prod darken-effect"
           style="grid-area: 8 / 1 / 9 / 2; background-color: rgb(255, 245, 157); color: rgb(0, 0, 0); border-radius: 2px;"
      >
        <div>
        </div>
      </div>
      <div class="pol__prod darken-effect"
           style="grid-area: 9 / 1 / 10 / 2; background-color: rgb(255, 245, 157); color: rgb(0, 0, 0); border-radius: 2px;"
      >
        <div>
        </div>
      </div>
      <div class="pol__prod darken-effect"
           style="grid-area: 10 / 1 / 11 / 2; background-color: rgb(255, 245, 157); color: rgb(0, 0, 0); border-radius: 2px;"
      >
        <div>
        </div>
      </div>
      <div class="pol__prod darken-effect"
           style="grid-area: 1 / 6 / 2 / 7; background-color: transparent; color: rgb(33, 33, 33); border-radius: 2px; font-weight: 400; line-height: 1.2;"
      >
        <div>
          Topping
        </div>
      </div>
      <div class="pol__prod darken-effect"
           style="grid-area: 2 / 6 / 3 / 7; background-color: rgb(206, 147, 216); color: rgb(0, 0, 0); border-radius: 2px;"
      >
        <div>
        </div>
      </div>
      <div class="pol__prod darken-effect"
           style="grid-area: 3 / 6 / 4 / 7; background-color: rgb(206, 147, 216); color: rgb(0, 0, 0); border-radius: 2px;"
      >
        <div>
        </div>
      </div>
      <div class="pol__prod darken-effect"
           style="grid-area: 4 / 6 / 5 / 7; background-color: rgb(206, 147, 216); color: rgb(0, 0, 0); border-radius: 2px;"
      >
        <div>
        </div>
      </div>
      <div class="pol__prod darken-effect"
           style="grid-area: 8 / 2 / 9 / 3; background-color: rgb(255, 171, 145); color: rgb(0, 0, 0); border-radius: 2px;"
      >
        <div>
        </div>
      </div>
      <div class="pol__prod darken-effect"
           style="grid-area: 9 / 2 / 10 / 3; background-color: rgb(255, 171, 145); color: rgb(0, 0, 0); border-radius: 2px;"
      >
        <div>
        </div>
      </div>
      <div class="pol__prod darken-effect"
           style="grid-area: 10 / 2 / 11 / 3; background-color: rgb(255, 171, 145); color: rgb(0, 0, 0); border-radius: 2px;"
      >
        <div>
        </div>
      </div>
      <pos-order-keyboard-stub>
      </pos-order-keyboard-stub>
      <dialog-text-filter-stub>
      </dialog-text-filter-stub>
      <dialog-product-search-result-stub>
      </dialog-product-search-result-stub>
    </div>
  </div>
  <dialog-choose-popup-modifier-stub>
  </dialog-choose-popup-modifier-stub>
  <g-overlay-stub value="false"
                  absolute="true"
                  opacity="0.25"
                  color="rgb(150, 150, 150)"
  >
    <g-icon-stub size="120">
      icon-print
    </g-icon-stub>
  </g-overlay-stub>
</div>
`;
